src/model_objects/component.rs

```diff
@@ -219,3 +455,323 @@ impl Declarations {
             .map(|(k, _)| k)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::JsonProjectLoader;
+    use std::collections::HashSet;
+    use test_case::test_case;
+    struct SetupContext {
+        test_comp: Component,
+        expected: HashSet<String>,
+    }
+
+    ///Simplifying the test process by loading a component in a separate function, instead of in each test
+    fn setup(comp_name: &str, expected: Vec<String>) -> SetupContext {
+        let mut project_loader = JsonProjectLoader::new_loader(PATH, crate::tests::TEST_SETTINGS);
+        project_loader.get_settings_mut().disable_clock_reduction = true;
+        let mut test_comp = project_loader.get_component(comp_name).unwrap().clone();
+        let expected: HashSet<String> = expected.into_iter().collect();
+        // Initialise clock usage structs for each clock in component.
+        test_comp.initialise_clock_usages();
+
+        SetupContext {
+            test_comp,
+            expected,
+        }
+    }
+
+    // File path to project for project_loader
+    // PopulateClocks is designed to test for additional edge cases
+    const PATH: &str = "samples/json/PopulateClocks";
+
+    #[test]
+    fn initialise_clock_usages() {
+        let context = setup("Update", vec![]);
+
+        assert!(
+            context.test_comp.clock_usages.contains_key("x")
+                && context.test_comp.clock_usages.contains_key("y")
+        );
+    }
+
+    // TODO: maybe update component names to reflect tests?
+    #[test_case("Machine",  vec!["E25".to_string(),"E29".to_string()],  true;  "Clock with usage in two guards")]
+    #[test_case("Machine",  vec!["E36".to_string(),"E45".to_string()],  false; "Clock with usage in two fake guards")]
+    #[test_case("Machine4", vec!["E1".to_string(),"E5".to_string()],    true;  "Clock with usage in two guards avoiding cherrypicking")]
+    #[test_case("Machine4", vec!["E36".to_string(),"E45".to_string()],  false; "Clock with usage in two fake guards avoiding cherrypicking")]
+    fn populate_usages_with_guards(comp_name: &str, expected_edges: Vec<String>, verdict: bool) {
+        // Instantiating variables used in all tests using the setup function above.
```

Comment:
Had to think twice whether the function was called "setup" or if the function was a "setup function".
Used the same quotation as for the comment 'Confirming edges where clock "y" exists.'
```suggestion
        // Instantiating variables used in all tests using the "setup" function above.
```